nanoqm/workflows/input_validation.py

```diff
@@ -256,6 +262,22 @@ def add_multiplicity(self) -> None:
                 s.specific.cp2k.force_eval.dft.multiplicity = self.general['multiplicity']
                 s.specific.cp2k.force_eval.dft.uks = ""
 
+    def add_functional_x(self) -> None:
+        """Add the keyword for the exchange part of the DFT functional: GGA or MGGA"""
+        for s in (
+            self.general[p] for p in [
+                'cp2k_settings_main',
+                'cp2k_settings_guess']):
+            s.specific.cp2k.force_eval.dft.xc.xc_functional[self.general['functional_x']] = {} 
+
+    def add_functional_c(self) -> None:
+        """Add the keyword for the correlation part of the DFT functional: GGA or MGGA"""
+        for s in (
```

Comment:
```suggestion
        self.general['functional_c'] is None:
            return
        for s in (
```
Skip the rest of the method if `functional_c` has not been specified.